### PR TITLE
update notion reader

### PR DESCRIPTION
**Pull request by jerryjliu**
improved notion reader based on some feedback:
- allow both page_ids and database_ids to be specified (before it was if/else database ids or page ids)
- allow a toggle to load all pages if neither is specified on `load_all_if_empty` while preserving backwards compat
- deal with duplicate page ids between databases/pages by maintaining a set

There's future work we can do to specifically load database pages but this should be ok for now

### CODE REVIEW

File: llama-index-integrations/readers/llama-index-readers-notion/llama_index/readers/notion/base.py [hunk 1]
```diff
@@ -190,40 +190,49 @@ def search(self, query: str) -> List[str]:
         return page_ids
 
     def load_data(
-        self, page_ids: List[str] = [], database_ids: Optional[List[str]] = None
+        self, 
+        page_ids: List[str] = [],
```
Comment (logan-markewich) [inline llama-index-integrations/readers/llama-index-readers-notion/llama_index/readers/notion/base.py:194]: Nit: we shouldn't be using an empty list as a default kwarg

File: llama-index-integrations/readers/llama-index-readers-notion/llama_index/readers/notion/base.py [hunk 2]
```diff
@@ -190,40 +190,49 @@ def search(self, query: str) -> List[str]:
...
-            for page_id in page_ids:
-                page_text = self.read_page(page_id)
-                docs.append(
-                    Document(
-                        text=page_text, id_=page_id, extra_info={"page_id": page_id}
-                    )
+                db_page_ids = self.query_database(database_id)
+                all_page_ids.update(db_page_ids)
+
+        all_page_ids_list = list(all_page_ids)
```
Comment (logan-markewich) [inline llama-index-integrations/readers/llama-index-readers-notion/llama_index/readers/notion/base.py:228]: Nit: we can iterate without converting to a list right?

**Review by logan-markewich — APPROVED** [review]
Lgtm, very minor comments

File: llama-index-integrations/readers/llama-index-readers-notion/llama_index/readers/notion/base.py (hunk 1 above)
Comment (jerryjliu) [reply llama-index-integrations/readers/llama-index-readers-notion/llama_index/readers/notion/base.py:194]: yeah i'm not sure how this decision was made, might leave it for now

**Pull request merged**